e2e/consultemplate/consultemplate_test.go

```diff
@@ -0,0 +1,547 @@
+		expectedEntries = append(expectedEntries, e)
+	}
+
+	// Ensure the direct service lookup has the new entries we expect.
+	err = waitForTaskFile(serviceLookupAllocID, "test", "${NOMAD_TASK_DIR}/service.conf",
+		func(out string) bool {
+			for _, entry := range expectedEntries {
+				if !strings.Contains(out, entry) {
+					return false
+				}
+			}
+			return true
+		}, nil)
+	must.NoError(t, err, must.Sprintf("expected: %v", expectedEntries))
+}
+
+func waitForTaskFile(allocID, task, path string, test func(out string) bool, wc *e2eutil.WaitConfig) error {
+	var err error
+	var out string
+	interval, retries := wc.OrDefault()
+
+	testutil.WaitForResultRetries(retries, func() (bool, error) {
+		time.Sleep(interval)
+		out, err = e2eutil.Command("nomad", "alloc", "exec", "-task", task, allocID, "sh", "-c", "cat "+path)
+		if err != nil {
+			return false, fmt.Errorf("could not cat file %q from task %q in allocation %q: %v",
+				path, task, allocID, err)
+		}
+		return test(out), nil
+	}, func(e error) {
+		err = fmt.Errorf("test for file content failed: got %#v\nerror: %v", out, e)
+	})
+	return err
+}
+
+// mustWaitTemplateRender is a helper that asserts a file has been rendered and
+// tests its contents
+func mustWaitTemplateRender(t *testing.T, allocID, path string, testFn func(string) error, timeout time.Duration, testSettings ...must.Setting) {
+	t.Helper()
+
+	must.Wait(t, wait.InitialSuccess(
+		wait.ErrorFunc(func() error {
+			out, err := e2eutil.Command("nomad", "alloc", "fs", allocID, path)
+			if err != nil {
+				return err
+			}
+			return testFn(out)
+		}),
+		wait.Gap(time.Millisecond*500),
+		wait.Timeout(timeout),
+	), testSettings...)
+}
+
+// mustWaitForStatusByGroup is similar to e2eutil.WaitForAllocStatus but maps
+// specific task group names to statuses without having to deal with specific
+// counts
+func mustWaitForStatusByGroup(t *testing.T, jobID, ns string,
+	expected map[string]string,
+	timeout time.Duration, testSettings ...must.Setting) {
+
+	t.Helper()
+
+	must.Wait(t, wait.InitialSuccess(
+		wait.ErrorFunc(func() error {
+			got, err := e2eutil.AllocsForJob(jobID, ns)
+			if err != nil {
+				return err
+			}
+			for _, row := range got {
+				group := row["Task Group"]
+				expectedStatus := expected[group]
+				gotStatus := row["Status"]
+				if expectedStatus != gotStatus {
+					return fmt.Errorf("expected %q to be %q, got %q",
+						group, expectedStatus, gotStatus)
+				}
+			}
+			return nil
+		}),
+		wait.Gap(time.Millisecond*500),
+		wait.Timeout(timeout),
+	), testSettings...)
+}
+
+func cleanupJob(t *testing.T, ns, jobID string) {
+	if os.Getenv("NOMAD_TEST_SKIPCLEANUP") == "1" {
+		return
+	}
+
+	t.Helper()
+	t.Cleanup(func() {
+		e2eutil.StopJob(jobID, "-purge", "-detach", "-namespace", ns)
+	})
+}
+
+func cleanupNamespace(t *testing.T, ns string) {
+	if os.Getenv("NOMAD_TEST_SKIPCLEANUP") == "1" {
```

Comment:
one advantage to `e2eutil.CleanupCommand()` is that it does this bit for ya (and checks for error, and optionally logs what it's up to if verbose is set)

Reply:
Done.